src/osiris_log.erl

```diff
@@ -1903,7 +1903,7 @@ update_retention(Retention,
 evaluate_retention(Dir, Specs) when is_list(Dir) ->
     % convert to binary for faster operations later
     % mostly in segment_from_index_file/1
-    evaluate_retention(list_to_binary(Dir), Specs);
+    evaluate_retention(unicode:characters_to_binary(Dir), Specs);
```

Comment:
You always convert to `unicode` = `utf8` even if the VM is running in `latin1` mode. I think it's probably fine here. Just want to add a note that
https://github.com/erlang/otp/blob/a1f235fc05653fa524fc7c4a352398af6ce0b364/lib/stdlib/src/filename.erl#L845-L851 uses `file:native_name_encoding()` as `OutEncoding`.